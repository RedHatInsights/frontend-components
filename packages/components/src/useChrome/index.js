export { default } from './useChrome';
export { default as useChrome } from './useChrome';
