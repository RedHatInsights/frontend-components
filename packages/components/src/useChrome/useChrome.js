import { useScalprum } from '@scalprum/react-core';

const useChrome = (selector) => {
    const state = useScalprum();
    let chrome = state.api?.chrome || {};
    chrome = {
        ...chrome,
        initialized: state.initialized
    };
    if (typeof selector === 'function') {
        return selector(chrome);
    }

    return chrome;
};

export default useChrome;
