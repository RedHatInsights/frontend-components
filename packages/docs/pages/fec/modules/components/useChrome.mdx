# useChrome

## useChrome
 **type**: `function`

```typescript
declare function useChrome(selector?: UseChromeSelector): void
```

## UseChromeSelector
 **type**: `function`

```typescript
declare function chromeState(chromeState: ChromeAPI): ChromeAPI
```
