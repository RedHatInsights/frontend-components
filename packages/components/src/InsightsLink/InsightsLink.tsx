import React from 'react';
import { Link, LinkProps } from 'react-router-dom';
import useChrome from '../useChrome';
import { buildInsightsPath } from '@redhat-cloud-services/frontend-components-utilities/helpers/urlPathHelpers';

interface InsightsLinkProps {
  to: LinkProps['to'];
  app: string;
  preview: boolean;
}

const InsightsLink: React.FC<InsightsLinkProps> = ({ to, app, preview, ...props }) => {
  const chrome = useChrome();
  const toPath = buildInsightsPath(chrome, app, to, preview);

  return (
    <Link to={toPath} {...props}>
      {props.children}
    </Link>
  );
};

export default InsightsLink;
