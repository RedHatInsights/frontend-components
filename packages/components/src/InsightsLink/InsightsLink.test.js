import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import InsightsLink from './InsightsLink';
import useChrome from '../useChrome';
jest.mock('../useChrome');

jest.mock('react-router-dom', () => ({
  Link: () => 'Mocked Link',
}));

describe('InsightsLink component', () => {
  describe('under /', () => {
    beforeEach(() => {
      useChrome.mockImplementation(() => ({
        isBeta: () => false,
        getApp: () => 'compliance',
        getBundle: () => 'insights',
      }));
    });

    it('should render a link with a given path', () => {
      const wrapper = mount(
        <InsightsLink to="/scappolicies">
          Link to Scap Policies under `/insights/compliance/...` will link to `/insights/compliance/scappolicies`
        </InsightsLink>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a link to /preview with a given path', () => {
      const wrapper = mount(<InsightsLink to="/scappolicies" preview />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a link to a different app in preview when given', () => {
      const wrapper = mount(
        <InsightsLink to={'/ID_OF_A_SYSTEM'} app="inventory" preview>
          Link Inventory sytems page under `/insights/compliance/systems` will link to `/preview/insights/inventory/ID_OF_A_SYSTEM`
        </InsightsLink>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('under /preview', () => {
    beforeEach(() => {
      useChrome.mockImplementation(() => ({
        isBeta: () => true,
        getApp: () => 'compliance',
        getBundle: () => 'insights',
      }));
    });

    it('should render a link to non /preview', () => {
      const wrapper = mount(
        <InsightsLink to={'/ID_OF_A_SYSTEM'} app="inventory" preview={false}>
          Link Inventory sytems page under `/preview/insights/compliance/systems` will link to `/insights/inventory/ID_OF_A_SYSTEM`
        </InsightsLink>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a /preview link', () => {
      const wrapper = mount(
        <InsightsLink to={{ pathname: '/scappolicies' }}>
          Link to Scap Policies under `/preview/insights/compliance` will link to `/insights/compliance/scappolicies`
        </InsightsLink>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
