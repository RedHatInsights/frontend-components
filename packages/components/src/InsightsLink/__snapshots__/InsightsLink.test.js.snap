// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsLink component under / should render a link to /preview with a given path 1`] = `
<InsightsLink
  preview={true}
  to="/scappolicies"
>
  <Link
    to="/preview/insights/compliance/scappolicies"
  >
    Mocked Link
  </Link>
</InsightsLink>
`;

exports[`InsightsLink component under / should render a link to a different app in preview when given 1`] = `
<InsightsLink
  app="inventory"
  preview={true}
  to="/ID_OF_A_SYSTEM"
>
  <Link
    to="/preview/insights/inventory/ID_OF_A_SYSTEM"
  >
    Mocked Link
  </Link>
</InsightsLink>
`;

exports[`InsightsLink component under / should render a link with a given path 1`] = `
<InsightsLink
  to="/scappolicies"
>
  <Link
    to="/insights/compliance/scappolicies"
  >
    Mocked Link
  </Link>
</InsightsLink>
`;

exports[`InsightsLink component under /preview should render a /preview link 1`] = `
<InsightsLink
  to={
    Object {
      "pathname": "/scappolicies",
    }
  }
>
  <Link
    to={
      Object {
        "pathname": "/insights/compliance/scappolicies",
      }
    }
  >
    Mocked Link
  </Link>
</InsightsLink>
`;

exports[`InsightsLink component under /preview should render a link to non /preview 1`] = `
<InsightsLink
  app="inventory"
  preview={false}
  to="/ID_OF_A_SYSTEM"
>
  <Link
    to="/insights/inventory/ID_OF_A_SYSTEM"
  >
    Mocked Link
  </Link>
</InsightsLink>
`;
